src/Microsoft.Content.Build.MerfToSDPForJava.Steps/TocYamlGenerator.cs

```diff
@@ -0,0 +1,65 @@
+﻿namespace Microsoft.Content.Build.MerfToSDPForJava.Steps
+{
+    using Microsoft.Content.Build.MerfToSDPForJava.Common;
+    using Microsoft.Content.Build.MerfToSDPForJava.Constants;
+    using Microsoft.Content.Build.MerfToSDPForJava.DataContracts.SDP;
+    using System;
+    using System.IO;
+    using System.Threading.Tasks;
+
+    public class TocYamlGenerator : IStep
+    {
+        public string StepName
+        {
+            get { return "TocToSDP"; }
+        }
+        public Task RunAsync(BuildContext context)
+        {
+            var config = context.GetSharedObject(Constants.Config) as ConfigModel;
+            if (config == null)
+            {
+                throw new ApplicationException(string.Format("Key: {0} doesn't exist in build context", Constants.Config));
+            }
+
+            string sdpYmlFolder = StepUtility.GetSDPYmlOutputPath(config.OutputPath);
+            if (!Directory.Exists(sdpYmlFolder))
+            {
+                throw new ApplicationException(string.Format("SDPYmlOutputPath: {0} doesn't exist.", sdpYmlFolder));
+            }
+
+            var inputPaths = config.InputPaths;
+            foreach (var inputPath in inputPaths)
+            {
+                try
+                {
+                    var tocFilePath = Path.Combine(inputPath, "toc.yml");
+                    if (!File.Exists(tocFilePath))
```

Comment:
if the toc.yml does not exists, thinking we could generate one as we have all the information we need, this do not need to be address now, I will create a task to track this work.

Reply:
toc.yml has to exist, and if it doesn't, it's wrong. because we generate it from upstream.